src/Infinities.jl

```diff
@@ -107,15 +107,17 @@ function mod(x::Real, ::Infinity)
     x
 end
 
+abstract type RealInfinity <: Real end
+struct PositiveInfinity <: RealInfinity end
+struct NegativeInfinity <: RealInfinity end
```

Comment:
Good, this also makes those two types into singleton types, which may allow for better code in a few places (avoiding allocations)